### PR TITLE
Call home deps missing

### PR DESCRIPTION
**Pull request by vinmazzi**
## Summary
This PR is based off of the install_docker branch.

This code introduces the creation of the REACHC_ID and the compileLog call when some of the dependencies are not installed.
callhome

### CODE REVIEW

File: reach [hunk 1]
```diff
@@ -115,10 +115,142 @@ run_s () {
...
+  START_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+  TRANSLATED_TEMPLATE=$(echo $CALL_HOME_TEMPLATE_DECODED | sed -e "s/{{REACHC_ID}}/$REACHC_ID/g" \
+  -e "s/{{START_TIME}}/${START_TIME}/g" \
+  -e 's/{{ELAPSED_TIME}}/none/g' \
+  -e "s/{{RESULT}}/${2} is not installed/g" \
+  -e 's/{{REACH_VERSION}}/0.1234/g' \
+  -e 's/{{CONNECTOR_MODE}}/none/g' \
+  -e 's/{{USING_VSSTUDIO_EXTENTION}}/none/g' \
+  -e "s/{{INITIATOR}}/${1}/g")
+  CALL_HOME_ENDPOINT="https://y9fr4e7fyl.execute-api.us-east-2.amazonaws.com/live/submit"
```
Comment (jeapostrophe) [inline reach:151]: Is that really the URL? What's wrong with https://github.com/reach-sh/reach-lang/blob/master/hs/src/Reach/Report.hs#L34

File: reach [hunk 2]
```diff
@@ -115,10 +115,142 @@ run_s () {
...
+wgImNvbm5lY3Rvck1vZGUiOiAie3tDT05ORUNUT1JfTU9ERX19IiwgInVzaW5nVmlzdWFsU\
+3R1ZGlvRXh0ZW5zaW9uIjogInt7VVNJTkdfVlNTVFVESU9fRVhURU5USU9OfX0iLCAiaW5p\
+dGlhdG9yIjogInt7SU5JVElBVE9SfX0iIH0="
+  CALL_HOME_TEMPLATE_DECODED=$(echo -n $CALL_HOME_ENCODED | base64 -d)
+  START_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+  TRANSLATED_TEMPLATE=$(echo $CALL_HOME_TEMPLATE_DECODED | sed -e "s/{{REACHC_ID}}/$REACHC_ID/g" \
+  -e "s/{{START_TIME}}/${START_TIME}/g" \
+  -e 's/{{ELAPSED_TIME}}/none/g' \
+  -e "s/{{RESULT}}/${2} is not installed/g" \
+  -e 's/{{REACH_VERSION}}/0.1234/g' \
```
Comment (jeapostrophe) [inline reach:147]: I think this version should be different; like 0.0.0 or something like that

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: reach [hunk 3]
```diff
@@ -115,10 +115,154 @@ run_s () {
...
+  if [ ! -e $REACH_ENV_FILE ]; then
+    echo "REACHC_ID=${PRE_REACHC_ID}" > $REACH_ENV_FILE
+  elif ( ! grep REACHC_ID $REACH_ENV_FILE > /dev/null 2>&1 ); then
+    echo "REACHC_ID=${PRE_REACHC_ID}" >> $REACH_ENV_FILE
+  fi
+}
+
+call_home () {
+
+  START_TIME=$1
```
Comment (jeapostrophe) [inline reach:136]: You take it as an argument

File: reach [hunk 4]
```diff
@@ -115,10 +115,154 @@ run_s () {
...
+  INITIATOR=$7
+
+  CALL_HOME_ENCODED="eyAidXNlcklkIjogInt7UkVBQ0hDX0lEfX0iLCAic3RhcnRUaW1l\
+IjogInt7U1RBUlRfVElNRX19IiwgImVsYXBzZWQiOiAie3tFTEFQU0VEX1RJTUV9fSIsICJ\
+yZXN1bHQiOiAie3tSRVNVTFR9fSIsICJ2ZXJzaW9uIjogInt7UkVBQ0hfVkVSU0lPTn19Ii\
+wgImNvbm5lY3Rvck1vZGUiOiAie3tDT05ORUNUT1JfTU9ERX19IiwgInVzaW5nVmlzdWFsU\
+3R1ZGlvRXh0ZW5zaW9uIjogInt7VVNJTkdfVlNTVFVESU9fRVhURU5USU9OfX0iLCAiaW5p\
+dGlhdG9yIjogInt7SU5JVElBVE9SfX0iIH0="
+  CALL_HOME_TEMPLATE_DECODED=$(echo -n $CALL_HOME_ENCODED | base64 -d)
+  START_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
```
Comment (jeapostrophe) [inline reach:151]: But then replace it

Also, I believe that we've switched to only using the date, not the time too

File: reach [hunk 5]
```diff
@@ -115,10 +115,154 @@ run_s () {
...
+
+  if [ ! -e $REACH_ENV_FILE ]; then
+    echo "REACHC_ID=${PRE_REACHC_ID}" > $REACH_ENV_FILE
+  elif ( ! grep REACHC_ID $REACH_ENV_FILE > /dev/null 2>&1 ); then
+    echo "REACHC_ID=${PRE_REACHC_ID}" >> $REACH_ENV_FILE
+  fi
+}
+
+call_home () {
+
```
Comment (jeapostrophe) [inline reach:135]: Blasphemous newline

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
I'd like @mattaudesse to do final approval

**Review by mattaudesse — CHANGES_REQUESTED** [review]

**Pull request closed without merging**